### PR TITLE
separated fake data into a function

### PR DESCRIPTION
**Pull request by jmfranck**

### CODE REVIEW

File: pyspecProcScripts/first_level/fake_data.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+from sympy import lambdify
+from pyspecdata import nddata, ndshape
+import numpy as np
+from numpy.random import normal
+def fake_data(
+        expression,
+        axis_coords,
+        signal_pathway,
+        direct = 't2',
+        SD_sigma = [0.05,0.003],
```
Comment (guinneak198) [inline pyspecProcScripts/first_level/fake_data.py:12]: will this be true for most data sets we create? I'm still iffy on how sigma/gaussians contribute to the fake data signal. Do you have references you can point me to to better connect these ideas?

File: pyspecProcScripts/first_level/fake_data.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        To enable *simulating echo-like data*, you can specify a direct axis
+        that starts at a negative number.
+        If you do this, the beginning of the axis will be re-set to 0 before returning.
+    SD_sigma: list of floats
+        Gives the Gaussian σ for the spectral density of the
+        frequency variation.
+        Typically, there are more than one gaussian terms used to
+        create the spectral density.
+
+        A small σ gives rise to slow fluctuations while a large σ
```
Comment (guinneak198) [inline pyspecProcScripts/first_level/fake_data.py:37]: is fluctuations the rainbowy phasing we see?

File: pyspecProcScripts/first_level/fake_data.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        gives rise to fast fluctuations.
+        The units of σ are in cycles per scan.
+    SD_amp: list of floats
+        Amplitudes associated with SD_sigma terms -- must be the
```
Comment (guinneak198) [inline pyspecProcScripts/first_level/fake_data.py:41]: you say associated with SD_sigma so is there an equation you use to figure out what amp is needed for what sigma?

File: pyspecProcScripts/first_level/fake_data.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    scale: float (default 100) 
+        amplitude of frequency variation
+    """
+    mysymbols = expression.atoms(sympy_symbol)
+    missing = (
+            set(str(j) for j in mysymbols)
+            - set(axis_coords.keys()))
+    assert len(missing) == 0, "all non-phase cycling symbols in your expression must have matching axis coordinates in the dictionary -- you are missing %s!" % str(missing)
+    thefunction = lambdify(mysymbols, expression, 'numpy')
+    clean_data = thefunction(*tuple(axis_coords[str(j)] for j in mysymbols))
```
Comment (guinneak198) [inline pyspecProcScripts/first_level/fake_data.py:55]: does the order of the axis_coords dictionary matter?

File: pyspecProcScripts/first_level/fl_dummy.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+"""provides object fl_dummy that can take the place of a figure list, but doesn't do anything.
+
+`fl=fl_dummy` is a better default than `fl=None`, because it doesn't require any conditionals
```
Comment (guinneak198) [inline pyspecProcScripts/first_level/fl_dummy.py:3]: if we said fl =fl where fl is figlist_var() that wouldn't require conditionals either would it? having trouble understanding why we wouldn't just say fl= None

File: examples/table_of_integrals.py [hunk 6]
```diff
@@ -20,60 +22,28 @@
...
+        ("t2" , nddata(r_[0:0.2:256j]-echo_time, "t2"))]),
+        {"ph1": 0, "ph2": 1})
+data.reorder(["ph1", "ph2", "vd"])
 fl.next("fake data -- time domain")
 fl.image(data)
-data.ft("t2", shift=True)
+data.ft("t2")
 # {{{ make it unitary again
-data /= sqrt(ndshape(data)["t2"]) * dt
+data /= sqrt(ndshape(data)["t2"]) * data.get_ft_prop('t2','dt')
```
Comment (guinneak198) [inline examples/table_of_integrals.py:43]: cool!

File: examples/table_of_integrals.py [hunk 7]
```diff
@@ -20,60 +22,28 @@
...
 # }}}
 fl.next("fake data -- freq domain")
 fl.image(data)
 # }}}
+# AG: implement your function on this data -- then also modify the symbolic expression to get enhancement, etc, and do that as well
```
Comment (guinneak198) [inline examples/table_of_integrals.py:48]: am I doing that in this pull request or a separate one after this gets merged?

**Review by guinneak198 — COMMENTED** [review]
AG reviewed and asked questions where relevant

File: pyspecProcScripts/first_level/fake_data.py (hunk 1 above)
Comment (jmfranck) [reply pyspecProcScripts/first_level/fake_data.py:12]: We are generating a random list of frequency offsets to model the field drift. These Gaussians are used to give those numbers a non-uniform spectral density.

File: pyspecProcScripts/first_level/fake_data.py (hunk 2 above)
Comment (jmfranck) [reply pyspecProcScripts/first_level/fake_data.py:37]: No, that comes from the fact that the echo is not centered at zero￼. 

The frequency draft results in the fact that the center of the signal is wiggling back and forward￼

File: pyspecProcScripts/first_level/fake_data.py (hunk 3 above)
Comment (jmfranck) [reply pyspecProcScripts/first_level/fake_data.py:41]: No, the Sigma and associated amplitudes or some thing that I messed around with to get a frequency variation that looks reasonable￼

File: pyspecProcScripts/first_level/fake_data.py (hunk 4 above)
Comment (jmfranck) [reply pyspecProcScripts/first_level/fake_data.py:55]: Yes, it's explained in the docstring

File: pyspecProcScripts/first_level/fl_dummy.py (hunk 5 above)
Comment (jmfranck) [reply pyspecProcScripts/first_level/fl_dummy.py:3]: This avoids `if fl is not None:as

File: examples/table_of_integrals.py (hunk 7 above)
Comment (jmfranck) [reply examples/table_of_integrals.py:48]: Separate after is good

Comment (jmfranck) [reply examples/table_of_integrals.py:48]: But you could create it as a branch off of this one.

File: pyspecProcScripts/first_level/fl_dummy.py (hunk 5 above)
Comment (guinneak198) [reply pyspecProcScripts/first_level/fl_dummy.py:3]: so is fl still a kwarg for functions?

File: examples/table_of_integrals.py (hunk 7 above)
Comment (guinneak198) [reply examples/table_of_integrals.py:48]: I need this for a couple other PRs/examples so I will just wait till this is merged.

**Review by guinneak198 — COMMENTED** [review]
looks good to me to merge!!!

**Pull request merged**